editor/code_editor.cpp

```diff
@@ -92,6 +92,13 @@ void FindReplaceBar::_notification(int p_what) {
 	} else if (p_what == NOTIFICATION_VISIBILITY_CHANGED) {
 
 		set_process_unhandled_input(is_visible_in_tree());
+	} else if (p_what == EditorSettings::NOTIFICATION_EDITOR_SETTINGS_CHANGED) {
```

Comment:
This seems to be overkill to me. It will reload icons every single time settings are changed, even if those are not icon related settings. You should check what settings changed, and only reload if a relevant themeing one was modified.

Reply:
@akien-mga this does not reload the icons, i' ve added a check in editor themes to only reload/regenerate when needed, but yeah it gegenerates the theme, fonts and all styles, there might be a better solution

Reply:
@akien-mga  I needed to add those icons overrides.
Icons are saved in the controls themselves so changing the theme does not change the controls.
So I do a lookup and a set_icon for each control which saves the icon itself and does not load in NOTIFICAITON_DRAW.

But it is correct that there are a LOT of things happening now when changing the theme...
So it would be nice to have a better solution.
We could add another EditorSettings:: ENUM (`EDITOR_THEME_SETTINGS_CHAGNED`)
and than we call sent different notifications.
But since the editor Settings are so generic (just categories via strings that can appear by just one line somewhere...) It is hard to decide when to call which callback without hardcoding the strings that trigger it.

What about adding a static method to EditorSettings : `String EditorSettings::get_chagned_setting()`.
(may that already exists...)
than we could encapsulate as much as possible in the NOTIFICATION_EDITOR_SETTINGS_CHANGED cases so that only the needed things happen... it's some work though...

Reply:
Yeah, maybe that's outside of the scope of this PR and something to work on after it's merged.